internal/stores/hostdb.go

```diff
@@ -241,7 +274,45 @@ func (h *dbHost) BeforeCreate(tx *gorm.DB) (err error) {
 	return nil
 }
 
-func (e *dbBlocklistEntry) AfterCreate(tx *gorm.DB) (err error) {
+func (e *dbAllowlistEntry) AfterCreate(tx *gorm.DB) error {
+	// NOTE: the ID is zero here if we ignore a conflict on create
+	if e.ID == 0 {
+		return nil
+	}
+
+	params := map[string]interface{}{
+		"entry_id":    e.ID,
+		"exact_entry": publicKey(e.Entry),
+	}
+
+	// insert entries into the allowlist
+	switch tx.Config.Dialector.Name() {
```

Comment:
I'd remain consistent with the style we use in `isSQLite` where we check for the type instead of a string. 
You can also use `tx.Dialector` instead of `tx.Config.Dialector`.

Maybe it's even nicer to change `isSQLite` from a method to a function to make sure we can use this in hooks as well.